### ISSUE TITLE
Query selector, use of `~` in `:not()` throws

### ISSUE THREAD

**Issue by Banou26** [dperini/nwsapi#55]
```js
import { JSDOM } from 'jsdom'

const doc = new JSDOM(`
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
</head>
<body>
</body>
</html>
`).window.document

doc.querySelector('.js-sns-icon-container ~ h2 ~ h2 ~ .spaceit_pad:not(.js-sns-icon-container ~ h2 ~ h2 ~ h2 ~ .spaceit_pad)')
```
throws

```
'.js-sns-icon-container ~ h2 ~ h2 ~ .spaceit_pad:not(.js-sns-icon-container ~ h2 ~ h2 ~ h2 ~ .spaceit_pad)' is not a valid selector
```

Trying to query for `.js-sns-icon-container ~ h2 ~ h2 ~ .spaceit_pad:not(.js-sns-icon-container ~ h2 ~ h2 ~ h2 ~ .spaceit_pad)` in browsers work completly fine (Tried in chrome and FF).

**Comment by cryosis7** [comment]
I am experiencing a similar issue with `table:not(table table)` i.e. All tables that are not descendants of another table.
My difference was that instead of throwing an error, it would not parse the selector in the `:not` parameters as a complex selector (As is permitted in [CSS Selectors Level 4](https://drafts.csswg.org/selectors-4/#negation))  
I observed the `:not` would only parse the last tag, for instance if my selector was `ul:not(p ul)` it would parse it as `ul:not(ul)`

**Comment by slavaleleka** [comment]
the same with adjacent sibling combinator `+` inside :not() pseudo-class arg
`div:not(h1 + p)`

there is no error [in browser](https://user-images.githubusercontent.com/45171506/208148027-e78591d7-f3bd-4a93-8d09-ad197453cc58.png)

**Comment by svarzee** [comment]
I came across the same issue. It seems selector validation is broken. If you really need it to work you can disable validation, keep in mind it is an ugly hack though:

In `node_modules/nwsapi/src/nwsapi.js` replace:
`reValidator = RegExp(standardValidator, 'g');`
with
`reValidator = RegExp('.*', 'g');`

**Comment by svarzee** [comment]
The issue shouldn't be closed, the problem wasn't fixed. Validation still fails for valid selectors mentioned in this issue.

**Comment by dperini** [comment]
@svarzee 
you are correct I should at least have pushed the code to fix this before closing.
However just for you to know I already have the code for that and I am in the process to get this fixed.

**Comment by dperini** [comment]
@Banou26 
I pushed a commit for the problem you have submitted (b9da5347250ca7c9796f70c86189dfbeacb9aa18) hope it fixes your issue.
It could be of help if you can check the github master repository and confirm the fix.
Thank you for your time and help.

**Comment by dperini** [comment]
Verified completed.
Closing issue.
